### PR TITLE
feat: get blockNumber from Ethers.js event data

### PR DESCRIPTION
**Pull request by bajtos**
I found a way how to access the ContractEventPayload from the contract event listener function. By getting the block number from the event data, we don't have to make another RPC call to find the block number via potentially expensive & slow `eth_getLogs`.

Very loosely related to:
- https://github.com/filecoin-station/spark-api/issues/311

### CODE REVIEW

**Comment by bajtos** [comment]
This pull request is based on https://github.com/filecoin-station/spark-api/pull/310 and it's blocked until we land that refactor.

**Review by juliangruber — APPROVED** [review]

File: lib/round-tracker.js
Comment (bajtos) [inline lib/round-tracker.js:42]: This seems to work very well. 👍🏻 We are advancing rounds, and the warning is not printed.

```
2024-05-21T08:56:31Z app[5683210c621628] cdg [info]Mapped 0x8460766Edc62B525fc1FA4D628FC79229dC73031 IE round index 7310n to SPARK round number 12733n (start epoch: 3932512)
2024-05-21T08:56:33Z app[5683210c621628] cdg [info]SPARK round started: 12733n (epoch: 3932512)
2024-05-21T08:56:33Z app[9185e760f41658] cdg [info]Mapped 0x8460766Edc62B525fc1FA4D628FC79229dC73031 IE round index 7310n to SPARK round number 12733n (start epoch: 3932512)
2024-05-21T08:56:33Z app[9185e760f41658] cdg [info]SPARK round started: 12733n (epoch: 3932512)
2024-05-21T08:56:33Z app[3d8dd965ae2998] cdg [info]Mapped 0x8460766Edc62B525fc1FA4D628FC79229dC73031 IE round index 7310n to SPARK round number 12733n (start epoch: 3932512)
2024-05-21T08:56:33Z app[3d8dd965ae2998] cdg [info]SPARK round started: 12733n (epoch: 3932512)
2024-05-21T08:56:34Z app[48ed67dc0e0308] cdg [info]Mapped 0x8460766Edc62B525fc1FA4D628FC79229dC73031 IE round index 7310n to SPARK round number 12733n (start epoch: 3932512)
2024-05-21T08:56:34Z app[48ed67dc0e0308] cdg [info]SPARK round started: 12733n (epoch: 3932512)
```

**Pull request merged**
